gobblin-runtime/src/test/java/org/apache/gobblin/runtime/FsDatasetStateStoreTest.java

```diff
@@ -181,6 +183,33 @@ public void testGetPreviousDatasetStatesByUrns() throws IOException {
     Assert.assertEquals(datasetState.getDuration(), 1000);
   }
 
+  /**
+   * Loading previous but corrupted statestore
+   *
+   * Specifically the example used here is the state store generated from previous gobblin-kafka version without
+   * changing the package name into apache-intialized.
+   *
+   * Should expect at least exception thrown.
+   * @throws IOException
+   */
+  @Test
+  public void testGetPreviousDatasetStatesByUrnsForFailure() throws IOException{
+    String JOB_NAME_FOR_INCOMPATIBLE_STATE_STORE = "test_failing_job";
+
+    FsDatasetStateStore _fsDatasetStateStore =
+        new FsDatasetStateStore(ConfigurationKeys.LOCAL_FS_URI,
+            "gobblin-runtime/src/test/resources/datasetState");
+
+    try {
+      Map<String, JobState.DatasetState> datasetStatesByUrns =
+          _fsDatasetStateStore.getLatestDatasetStatesByUrns(JOB_NAME_FOR_INCOMPATIBLE_STATE_STORE);
+
+      fail("Failed as the problem doesn't throw expected exception here.");
```

Comment:
Please change this static invocation to be consistent with other places in the code ie. use Assert.fail( .. ) instead of static import of the call and invoking fail( .. )